### DISCUSSION TITLE
Correct way to inject grpc request interceptor

### DISCUSSION THREAD

**Discussion by dimasm61** [camunda-community-hub/zeebe-client-csharp#629] [Q&A] [answered]
After the update zb-client (1.3.0->2.4.0) new error appeared
```
Grpc.Core.RpcException: Status(StatusCode="Internal", 
Detail="Error starting gRPC call. HttpRequestException: The HTTP request headers length exceeded the server limit of 8192 bytes.",
DebugException="System.Net.Http.HttpRequestException: The HTTP request headers length exceeded the server limit of 8192 bytes.")
 ---> System.Net.Http.HttpRequestException: The HTTP request headers length exceeded the server limit of 8192 bytes.
   at System.Net.Http.Http2Connection.WriteHeaders(HttpRequestMessage request, ArrayBuffer& headerBuffer)
   at System.Net.Http.Http2Connection.SendHeadersAsync(HttpRequestMessage request, CancellationToken cancellationToken, Boolean mustFlush)
...
--- End of inner exception stack trace ---
   at Zeebe.Client.Impl.Commands.CompleteJobCommand.Send(Nullable`1 timeout, CancellationToken token)
   at Zeebe.Client.Impl.Commands.CompleteJobCommand.Send(CancellationToken cancellationToken)
```

In some cases we have a really large header string. The header send automatically. This is not required in the grpc request.

As I understand It, I need implement a grpc request interceptor where I can clean this header.

Does anyone know correct way to inject grpc request interceptor for zeebe-client-csharp?

I see how this is [implemented](https://github.com/camunda-community-hub/zeebe-client-csharp/blob/95b69603fcf3c98d1fddc564efd63eb459eaa60e/Client/ZeebeClient.cs#L100) in ZeebeClient class, but here averything is closed to modifications :(

**Comment by ChrisKujawa** [comment]
I wonder why you experienced this issue. Do you see what is part of such headers?

**Reply by dimasm61** [reply, marked as answer]
Hi

I done debugging with old and new version. It seems that an interceptor in the zeebe-client will not solve my issue.

We use OpenTelemetry for observability. With OpenTelemetry we use the Propagators mechanism to forward traces through headers. And it looks like it processes packets after the zeebe-client interceptor.

In the changes from version 1.3 to version 2.4 in the zeebeClient.cs class file, for the channelToGateway field type has changed (Channel->GrpsChannel).

With Channel type, Propagators mechanism does not work for grpc packets. In workers, we pass traces through variables, not through headers. The propagator does not augment packets.

With GrpsChannel type the behavior changed, the OpenTelemetry Propagator started working for grps packets. And in the zeebe workers service, headers began to be added to grps packets, sometimes very long.

The problem can not solved in the zeebe-client, the discussion can be closed.

**Reply by ChrisKujawa** [reply]
Thanks for coming back and clarifying 👍🏼
